mod/pages/start.php

```diff
@@ -357,33 +358,40 @@ function pages_write_permission_check($hook, $entity_type, $returnvalue, $params
 }
 
 /**
- * Extend container permissions checking to extend can_write_to_container for write users.
+ * Check container permissions
+ * For BC, we want to allow users to write 'page' objects to 'user' and 'group' entities.
  *
- * @param string $hook
- * @param string $entity_type
- * @param bool   $returnvalue
- * @param array  $params
+ * @param string $hook   "container_permissions_check"
+ * @param string $type   "object"
+ * @param bool   $return Permission
+ * @param array  $params Hook params
  *
  * @return bool
  */
-function pages_container_permission_check($hook, $entity_type, $returnvalue, $params) {
-	if (elgg_get_context() != "pages") {
-		return null;
+function pages_container_permission_check($hook, $type, $return, $params) {
+	
+	if (!elgg_in_context('pages')) {
+		return;
 	}
-	$container = elgg_get_page_owner_entity();
-	if ($container && $container->canWriteToContainer(0, 'object', 'page')) {
-		return true;
+
+	$user = elgg_extract('user', $params);
+	$container = elgg_extract('container', $params);
+	$subtype = elgg_extract('subtype', $params);
+
+	if ($type !== 'object' || $subtype !== 'page') {
+		return;
 	}
-	if ($page_guid = get_input('page_guid', 0)) {
+
+	if ($page_guid = get_input('page_guid')) {
 		$entity = get_entity($page_guid);
-	} elseif ($parent_guid = get_input('parent_guid', 0)) {
+	} elseif ($parent_guid = get_input('parent_guid')) {
 		$entity = get_entity($parent_guid);
 	}
-	if (isset($entity) && pages_is_page($entity)) {
```

Comment:
Ultimately canEdit = canWriteToContainer when it comes to adding new subpages to pages.